Documentation/tutorial/howto_filter_out_unwanted_apis.md

```diff
@@ -0,0 +1,124 @@
+How-to: Filter Out Unwanted APIs 
+===============================
+
+A filter configuration file is a YAML formatted file(.yml). You could filter out unwanted APIs by providing a filter configuration file and specifying its path.
+
+## Specifying filter configuration file path
+The path of the configuration file could be specified by the following two ways. Option1 could overwrite option2.  
+1. docfx.exe metadata command argument.  
+`docfx.exe metadata -filter <path relative to baseDir or absolutepath>`
+2. docfx.json metadata section `filter` property.
+ 
+   **Sample**
+   ```
+   {
+     "metadata": [
+       {
+         "src": [
+           {
+             "files": [
+               "src/**.csproj"
+             ],
+             "exclude": [
+               "**/bin/**",
+               "**/obj/**"
+             ]
+           }
+         ],
+         "dest": "obj/api",
+         "filter": "filterConfig.yml"
+       }
+     ]
+   }
+   ```
+
+## The format of the filter configuration file  
+You could specify `apiRules` with a list of `exclude` or `include` rules.
+
+> *Note*
+
+> The rules would be executed sequentially and the matching process would stop once one rule is matched. 
+> Namely, you need to put the most detailed rule in the front.
+
+> If no rule is matched, the API would be included by default.
+
+
+  * `exclude` or `include` APIs by matching their uid with the Regex `uidRegex`.  
+  
+  The below sample would exclude all apis whose uid start with 'Microsoft.DevDiv' except those that start with 'Microsoft.DevDiv.SpecialCase'.  
+  ```
+    - include:
+        uidRegex: ^Microsoft\.DevDiv\.SpecialCase
+    - exclude:
+        uidRegex: ^Microsoft\.DevDiv
+  ```
+
+  * `exclude` or `include` APIs by matching its `type`, this is often combined with `uidRegex`.  
+  
+  Supported `type`:
+   * Namespace
+   * Type
+   * Class
+   * Struct
+   * Enum
+   * Interface
+   * Delegate
+   * Member
+   * Event
+   * Field
+   * Method
+   * Property
+  
+  > *Note*
+
+  > If a namespace is excluded, all types/members defined in the namespace would be excluded, too.
```

Comment:
point out that our namespace is flattened. So that it does not influence other namespaces